### PR TITLE
Refactor code the previous code

### PR DESCRIPTION
**Pull request by LoviMadaan**
### In this branch, We followed the pair-programming approach to meet the project requirements and changed the following :

- [ ] Made one modules folder for modular ruby
- [ ] Used YAGNI, KISS, and DRY rules and Ruby style guide are respected.
- [ ] Follow SOLID principles.
- [ ] Pay special attention to the "Single-responsibility Principle".
- [ ] Extract some pieces of logic into a separate class(es).

### CODE REVIEW

File: modules/create_book.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+
+    book = Book.new(title, author)
+    books << book
+    puts 'Book created succesfully.'
+    puts "\n"
+  end
```
Comment (mikethreels) [inline modules/create_book.rb]: - [ ] All input and input validation requests should've been placed in the main.rb file and/or classes/methods which are responsible for requesting or validating input. please have another look at [this lesson](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) which explains the best practices and the solid principle.

File: modules/create_persons.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    print 'Name: '
+    name = gets.chomp
+    print 'Specialization: '
+    specialization = gets.chomp
+    teacher = Teacher.new(name, age, specialization)
+    @people << teacher
+    puts 'Person created successfully'
+    puts "\n"
+  end
+end
```
Comment (mikethreels) [inline modules/create_persons.rb]: - [ ] All input and input validation requests should've been placed in the main.rb file and/or classes/methods which are responsible for requesting or validating input. please have another look at [this lesson](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) which explains the best practices and the solid principle.

File: modules/create_rental.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    if person_choice < 0 || person_choice > @people.length
+      puts 'Invalid person selection.'
+      puts "\n"
+      return
+    end
+
+    selected_person = @people[person_choice - 1]
+
+    print 'Date: '
+    rental_date = gets.chomp
```
Comment (mikethreels) [inline modules/create_rental.rb]: - [ ] All input and input validation requests should've been placed in the main.rb file and/or classes/methods which are responsible for requesting or validating input. please have another look at [this lesson](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) which explains the best practices and the solid principle.

File: modules/list_rentals.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+      return
+    end
+    puts 'Rentals:'
+    rentals_for_person = @rentals.select { |rental| rental.person == person }
+    rentals_for_person.each do |rental|
+      puts "Date: #{rental.date},  Book \"#{rental.title}\"  by #{rental.author}"
+    end
+    puts "\n"
+  end
+end
```
Comment (mikethreels) [inline modules/list_rentals.rb]: - [ ] All input and input validation requests should've been placed in the main.rb file and/or classes/methods which are responsible for requesting or validating input. please have another look at [this lesson](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) which explains the best practices and the solid principle.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi team,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Good job, starting to refactor the code.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: modules/create_persons.rb (hunk 2 above)
Comment (LoviMadaan) [reply modules/create_persons.rb]: @mikethreels Can you please elaborate? I am not getting what should i do?

**Review by Davidosky007 — APPROVED** [review]
Hi @LoviMadaan 👋 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* SOLID principle followed 👍 
* Ruby code refactored 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**